src/mame/dataeast/simpl156.h

```diff
@@ -15,16 +15,19 @@
 class simpl156_state : public driver_device
 {
 public:
-	simpl156_state(const machine_config &mconfig, device_type type, const char *tag)
-		: driver_device(mconfig, type, tag),
+	simpl156_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
 		m_maincpu(*this, "maincpu"),
 		m_deco_tilegen(*this, "tilegen"),
 		m_eeprom(*this, "eeprom"),
-		m_okimusic(*this, "okimusic") ,
-		m_mainram(*this, "mainram"),
-		m_systemram(*this, "systemram"),
+		m_okimusic(*this, "okimusic"),
 		m_sprgen(*this, "spritegen"),
-		m_palette(*this, "palette") { }
+		m_palette(*this, "palette"),
+		m_rowscroll(*this, "rowscroll_%u", 1U, 0x800U, ENDIANNESS_LITTLE),
+		m_mainram(*this, "mainram", 0x4000U, ENDIANNESS_LITTLE),
+		m_systemram(*this, "systemram"),
+		m_spriteram(*this, "spriteram", 0x1000U, ENDIANNESS_LITTLE)
```

Comment:
I’m not sure what size `m_rowscroll` is supposed to be.  Before this PR, it was allocating 0x800/2 16-bit words, i.e. 0x400 16-bit words or 0x800 bytes.  I’ve made it not reduce that.

However, if you look at dataeast/simpl156.cpp, it maps the handlers across a range of 0x2000 byte addresses, or 0x800 32-bit word addresses.  There’s one 16-bit row scroll word at each 32-bit CPU word address, so to avoid buffer overruns, it needs to be *0x800 16-bit words* rather than bytes.

Currently, accessing the second half of either row scroll area causes a buffer overrun (e.g. 0x161000-0x161fff and 0x165000-0x165fff in Joe and Mac Returns, or 0x3e1000-0x3e1fff and 0x3e5000-0x3e5fff in Magical Drop).

The solution is probably one of:
* Double the row scroll RAM size
* Mirror the second half of each row scroll area
* Halve the length of each row scroll area